CMakeLists.txt

```diff
@@ -151,57 +152,84 @@ if(ENABLE_ALL_DOC)
     set(ENABLE_USER_DOC ON)
 endif()
 
-#
-# Setup config default values, definitions, and configure files etc.
-#
-# These values are used to configure the general config file and are
-# picked up by the ConfigDefaults class to be used for default config
-# values.
-#
-
 SET(SYS_CONFIG_DIR ${CMAKE_INSTALL_FULL_SYSCONFDIR}/softwarecontainer/)
 add_feature_info("Config directory" TRUE "Configuration files will be installed in ${SYS_CONFIG_DIR}")
 
-# Since the general config file needs a lower case string as value
-# for the booleans, we need to set specific variables for these
-set(KEEP_CONTAINERS_ALIVE FALSE)
-if(${KEEP_CONTAINERS_ALIVE})
-    set(KEEP_CONTAINERS_ALIVE_VALUE_STRING "true")
-else(${KEEP_CONTAINERS_ALIVE})
-    set(KEEP_CONTAINERS_ALIVE_VALUE_STRING "false")
-endif()
 
-set(USE_SESSION_BUS FALSE)
-if(${USE_SESSION_BUS})
-    set(USE_SESSION_BUS_VALUE_STRING "true")
-else(${USE_SESSION_BUS})
-    set(USE_SESSION_BUS_VALUE_STRING "false")
-endif()
+#
+# Setup configs that goes into the main config file. Macros for configs are defined in Config.cmake
+#
+
+# Used as prefix to config keys that belong to the SoftwareContainer group
+set(SC_CONFIG_GROUP "SC")
+
+add_boolean_config(${SC_CONFIG_GROUP}
+                   KEEP_CONTAINERS_ALIVE
+                   FALSE
+                   "false")
```

Comment:
The default value is used for the definition that is used in the code later. The CMake "types" FALSE and TRUE will work for that, but they can't be used for the config file value which needs to be a string set to "true" or "false". I'll add some comment about this so it's more clear why it's done like this.